tests/unit/labels/MappedLabels_unittest.cc

```diff
@@ -0,0 +1,117 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Heiko Strathmann
+ */
+#include <gtest/gtest.h>
+#include <shogun/labels/MappedLabels.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/lib/SGMatrix.h>
+#include <shogun/classifier/svm/LibSVM.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/labels/MappedBinaryLabels.h>
+#include <shogun/kernel/GaussianKernel.h>
+
+
+#include "environments/LinearTestEnvironment.h"
+
+extern LinearTestEnvironment* linear_test_env;
+
+using namespace shogun;
+
+class MappedLabelsFixture : public ::testing::Test
+{
+protected:
+	void SetUp()
+	{
+		X = SGMatrix<float64_t>(1, 4);
+		feats = new CDenseFeatures<float64_t>(X);
+		SG_REF(feats);
+		svm = new CLibSVM();
+		SG_REF(svm);
+		kernel = new CGaussianKernel();
+		SG_REF(kernel);
+		svm->set_kernel(kernel);
+		result = nullptr;
+		labs = nullptr;
+	}
+
+	void TearDown()
+	{
+		SG_UNREF(feats);
+		SG_UNREF(labs);
+		SG_UNREF(svm);
+		SG_UNREF(kernel);
+		SG_UNREF(result);
+	}
+
+public:
+	void train()
+	{
+		svm->set_labels(labs);
+		svm->train(feats);
+	}
+
+	CLabels* apply()
+	{
+		result = svm->apply();
+		return result;
+	}
+
+	SGMatrix<float64_t> X;
+	CDenseFeatures<float64_t>* feats;
+	CDenseLabels* labs;
+	CMulticlassLabels* labels_mc;
+	CLibSVM* svm;
+	CGaussianKernel* kernel;
+	CLabels* result;
+};
+
+TEST_F(MappedLabelsFixture, mc_as_bin)
```

Comment:
see here for how it works for the user providing multiclass labels with only two (arbitrary!) values as binary labels